specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2018-02-01-preview/containerregistry_build.json

```diff
@@ -0,0 +1,2136 @@
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "BuildDefinitions_List": {
+            "$ref": "./examples/BuildDefinitionsList.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerRegistry/registries/{registryName}/buildDefinitions/{buildDefinitionName}/builds": {
+      "get": {
+        "tags": [
+          "BuildDefinitions"
+        ],
+        "summary": "Get all the builds for a given build definition.",
+        "operationId": "BuildDefinitions_ListBuilds",
+        "parameters": [
+          {
+            "name": "$filter",
+            "in": "query",
+            "description": "OData Filter options",
+            "required": false,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/registryNameParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/buildDefinitionNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The request was successful; the request was well-formed and received properly.",
+            "schema": {
+              "$ref": "#/definitions/BuildListResult"
+            }
+          }
+        },
+        "x-ms-odata": "#/definitions/BuildFilter",
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "BuildDefinitions_ListBuilds": {
+            "$ref": "./examples/BuildDefinitionsListBuilds.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerRegistry/registries/{registryName}/buildDefinitions/{buildDefinitionName}": {
+      "get": {
+        "tags": [
+          "BuildDefinitions"
+        ],
+        "summary": "Get the properties of a specified build definition.",
+        "operationId": "BuildDefinitions_Get",
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/registryNameParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/buildDefinitionNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The request was successful; the request was well-formed and received properly.",
+            "schema": {
+              "$ref": "#/definitions/BuildDefinition"
+            }
+          },
+          "404": {
```

Comment:
We didn't have negative cases in past swaggers and I would suggest not have them but if you wish to have 404, please add it to other APIs too.